lib/util/s3_setup.js

```diff
@@ -33,6 +32,10 @@ module.exports.detect = function(opts, config) {
       }
     }
   }
+  // If we are using s3ForcePathStyle the bucket is part of the http object path
+  // (but not the S3 key prefix path). If we aren't this isn't as much of a consideration.
+  const bucket_path = config.s3ForcePathStyle ? `/${config.bucket}/` : '/';
```

Comment:
Moving the below the if/else structure _does_ skip an early `return` on line 21.  I _think_ that early return is acting in light of an error condition so not having `config.prefix` set for it is fine, but I'm not 100% sure.